proto/eventlog.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+import "proto/context.proto";
+
+package eventlog;
+
+option go_package = "proto;eventlog";
+
+message ChunkRange {
+	// Negative start chunk indicates a chunk at index len(chunks) - start_chunk
+	// e. g. -1 is the last chunk, -2 is the second to last, etc.
+	int32 start_chunk = 1;
+	int32 num_chunks = 2;
+}
+
+message LineRange {
+	// Negative offset indicates that the offset is relative to the end
+	int64 num_lines = 2;
+}
+
+message ByteRange {
+	// Negative offset indicates that the offset is relative to the end
+	int64 offset = 1;
+	int64 num_bytes = 2;
+}
+
+message Range {
+	// Reverse means that we are reversing the order we read data in (though
+  // not reversing the order of data within a response); in it set, all offsets
+  // or start chunks are relative to the end of the data and all sizes are how
+  // much to read before the given offset
+	bool reverse = 1;
+	// Chunk to start from (leave 0 to use absolute offset)
+	int32 start_chunk = 2;
+	// Relative offset after the start chunk to start from
+	int64 offset = 3;
+	oneof size {
+		int32 num_chunks = 4; 
+		int64 num_lines = 5; 
+		int64 num_bytes = 6; 
+	}
+}
+
+message GetEventLogRequest {
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // The invocation identifier itself.
+  string invocation_id = 2;
+	
+	// The range of the event log to retrieve
+	Range range = 3;
+
+}
```

Comment:
My plan was only to implement support the chunk ranges at the start but design it this way so that it was easy to add other options later; wdyt? I figured this might be a good idea because if we're letting timeouts form chunks (which we probably want now that the plan is no longer to use the invocation to store the tail) we can end up with really short chunks sometimes, which could be a weird/bad user experience, but still very good for getting started (and also more performant), and then later we can add linewise requests. Maybe this is solvable by including a minimum number of lines in the proto and just retrieve whole chunks until that minimum is satisfied?

Reply:
Ack -- I get what you mean, but I think it's a little premature. HTTP reqs are fast and if we have to do a few back-to-back because we got some small chunks, I don't expect it to be an issue. The server could also mask this by combining consecutive chunks if they are small and just changing the pointer to next-chunk.

Can we start with the simple design and revisit this if it's a problem?

Reply:
yup, no issue with me